### DISCUSSION TITLE
Generate client-side types for operations without clutter

### DISCUSSION THREAD

**Discussion by eurostar-fennec-cooper** [dotansimha/graphql-code-generator#4458] [General] [answered]
First of all: excellent library!
I'm not sure if this is already possible or not, but I've searched plugins/documentation and can't seem to find anything.

When generating types for operations only, it would be useful to be able to resrict the types generated only to those referenced by the given operations.

If I enable `onlyOperationTypes` it does not generate types used by my operations, however if I disable it, it generates every single type. It would be nice to be able to have a configuration option that allows us to generate types only if they're referenced by an operation.

This would allow for incredibly clean/tidy types for client-side usage.

Additionally, it would be nice when generating types for a client-side project, to retrict generation only to operations defined/used client-side (ie specified via generates:documents).

The goal is to be able to write canned queries in .graphql files for a client-side project and then only generate types for those without filling the types file with unused types.

**Comment by danielrearden** [comment, marked as answer]
Hi @eurostar-fennec-cooper! Thanks for submitting an issue.

We're currently tracking this feature request [here](https://github.com/dotansimha/graphql-code-generator/issues/4373).
